packages/gestalt/src/Masonry/defaultLayout.ts

```diff
@@ -100,9 +102,14 @@ const defaultLayout =
         })
       : items.reduce<Array<any>>((acc, item) => {
           const positions = acc;
-          const height = measurementCache.get(item);
+          let height = measurementCache.get(item);
           let position;
 
+          if (renderLoadingState) {
+            // Height is a required key for _loadingStateItems
+            height = (item as { height: number }).height;
```

Comment:
lmk your thoughts here -- we could update the generic to be `T extends { height?: number }`, but that would require updating Masonry's generic to be the same and this felt like the shortest solution

Reply:
As mentioned above to me it sounds better to add a new second prop to have (items, loadingItems), that way you can keep the T type for items and the new {height: ..} structure for the loadingItems prop. Aldo with that you could add logic here to differentiate the loading items from the normal items

Reply:
marking this one as resolved since its a duplicate of https://github.com/pinterest/gestalt/pull/3637#discussion_r1680007162